constraints.txt

```diff
@@ -4,3 +4,4 @@ pywin32==225
 setuptools==49.6.0
 pyfakefs==4.1.0
 cvxpy==1.1.7
+scikit-learn>=0.24.2
```

Comment:
This isn't how the constraints file is used. This is used to pin extra constraints on top of the requirements, the normal use case is if a 2nd level dependency (ie a dependency of a dependency) has an issue that causes a problem we can use this file to pin it. It's not the place to specify a minimum version for an optional dependency.

Reply:
Thank you. I will restore it.

Reply:
Updated it